### PR TITLE
(CDAP-12662) Improve exception logging in TMS fetch handler

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java [hunk 1]
```diff
@@ -240,11 +251,14 @@ public void finished() throws Exception {
...
+      if (cause instanceof SocketException
+        || cause instanceof ClosedChannelException
+        || (cause instanceof IOException && KNOWN_IO_EXCEPTION_MESSAGES.contains(cause.getMessage()))) {
         // This can easily caused by client close connection prematurely. Don't want to flood the log.
-        LOG.debug("Socket exception raised when sending messages back to client", cause);
+        LOG.trace("Connection closed by client prematurely while sending messages back to client", cause);
       } else {
-        LOG.warn("Exception raised when sending messages back to client", cause);
+        // Use sampling logger to log to avoid flooding the log if there is any systematic failure
+        SAMPLING_LOG.warn("Exception raised when sending messages back to client", cause);
```
Comment (anew) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java:261]: would you still log it as trace(). The sampling logger would suppress some messages completely...

File: cdap-distributions/src/etc/cdap/conf.dist/logback-container.xml [hunk 2]
```diff
@@ -48,6 +48,12 @@
   <logger name="org.apache.twill" level="INFO"/>
   <logger name="org.apache.twill.internal.kafka.client.SimpleKafkaConsumer" level="WARN"/>
   <logger name="co.cask.cdap" level="INFO"/>
+  <!--
+    The netty http library logs an ERROR always if it is raised
+    not from user handler, which usually is due to network error.
+    Disabling the log to avoid flooding.
```
Comment (anew) [inline cdap-distributions/src/etc/cdap/conf.dist/logback-container.xml:54]: the comment is not clear. Then why would we not suppress a netty class logger?

Comment (chtyim) [reply cdap-distributions/src/etc/cdap/conf.dist/logback-container.xml:54]: I don't get this. The netty http class logger is being suppressed by setting level to OFF.

Comment (anew) [reply cdap-distributions/src/etc/cdap/conf.dist/logback-container.xml:54]: nvm

File: cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java (hunk 1 above)
Comment (chtyim) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java:261]: Added a trace

Comment (anew) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java:261]: Do you want to use the same message as above?

Comment (anew) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/server/FetchHandler.java:261]: ah they are different

**Comment by anew** [comment]
LGTM

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-RUT1420

**Comment by sreevatsanraman** [comment]
Checkstyle passed merging.

```INFO] CDAP Integration Test Framework .................... SUCCESS [  1.160 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:01 min
[INFO] Finished at: 2017-09-22T15:29:49-07:00
[INFO] Final Memory: 310M/873M
[INFO] ------------------------------------------------------------------------```

**Pull request merged**